core/deployment/src/main/java/io/quarkus/deployment/builditem/RunTimeConfigurationDefaultBuildItem.java

```diff
@@ -22,7 +22,7 @@ public final class RunTimeConfigurationDefaultBuildItem extends MultiBuildItem {
     public RunTimeConfigurationDefaultBuildItem(final String key, final String value) {
         Assert.checkNotNullParam("key", key);
         Assert.checkNotEmptyParam("key", key);
-        Assert.checkNotNullParam("value", value);
+        Assert.checkNotNullParam("value for key " + key, value);
```

Comment:
Good call! I wonder if we should get rid of these `Assert` calls and use proper `IllegalArgumentException`. That would allow for a more meaning full message.

But we can do it afterwards.